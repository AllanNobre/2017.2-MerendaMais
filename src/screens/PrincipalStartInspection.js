import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { TabViewAnimated, SceneMap, TabBar } from 'react-native-tab-view';
import StartPendingInspectionContainer from '../Containers/StartPendingInspectionContainer';
import StartExpiredInspectionContainer from '../Containers/StartExpiredInspectionContainer';
import Header from '../components/Header';
import StartAlreadyInspectionedInspectionContainer from '../Containers/StartAlreadyInspectionedInspectionContainer';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default class Principal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      index: 0,
      routes: [
        { key: 'first', title: 'Pendentes' },
        { key: 'second', title: 'Fiscalizados' },
        { key: 'third', title: 'Expirados' },
      ],
    };
  }

  handleIndexChange = index => this.setState({ index });

  renderFooter = props => <TabBar style={{ backgroundColor: '#FF9500', borderTopWidth: 1, borderBottomColor: 'black' }} tabStyle={{ paddingHorizontal: -0 }} {...props} />;

  renderHeader = props => <Header title={'Visitas Agendadas'} backButton {...props} />

  renderScene = SceneMap({
    first: StartPendingInspectionContainer,
    second: StartAlreadyInspectionedInspectionContainer,
    third: StartExpiredInspectionContainer,
  });

  render() {
    return (
      <TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderHeader={this.renderHeader}
        renderFooter={this.renderFooter}
        renderScene={this.renderScene}
        onIndexChange={this.handleIndexChange}
      />
    );
  }
}
