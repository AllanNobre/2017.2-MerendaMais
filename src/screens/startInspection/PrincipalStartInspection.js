import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { TabViewAnimated, SceneMap, TabBar } from 'react-native-tab-view';
import { Actions } from 'react-native-router-flux';
import StartPendingInspectionContainer from '../../Containers/startInspection/StartPendingInspectionContainer';
import StartExpiredInspectionContainer from '../../Containers/startInspection/StartExpiredInspectionContainer';
import Header from '../../components/Header';
import StartAlreadyInspectionedInspectionContainer from '../../Containers/startInspection/StartAlreadyInspectionedInspectionContainer';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default class Principal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      index: 0,
      routes: [
        { key: 'first', title: 'Pendentes' },
        { key: 'second', title: 'Fiscalizados' },
        { key: 'third', title: 'Expirados' },
      ],
    };
  }

  handleIndexChange = index => this.setState({ index });

  renderFooter = props => <TabBar style={{ backgroundColor: '#FF9500', borderTopWidth: 1, borderBottomColor: 'black' }} tabStyle={{ paddingHorizontal: -0 }} {...props} />;

  renderHeader = props => <Header title={'Visitas Agendadas'} backButton {...props} backTo={() => Actions.mainScreen()} />

  renderScene = SceneMap({
    first: StartPendingInspectionContainer,
    second: StartAlreadyInspectionedInspectionContainer,
    third: StartExpiredInspectionContainer,
  });

  render() {
    return (
      <TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderHeader={this.renderHeader}
        renderFooter={this.renderFooter}
        renderScene={this.renderScene}
        onIndexChange={this.handleIndexChange}
      />
    );
  }
}
