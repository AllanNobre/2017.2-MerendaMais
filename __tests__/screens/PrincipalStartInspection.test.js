import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// import configureStore from 'redux-mock-store';
import PrincipalStartInspection from '../../src/screens/PrincipalStartInspection';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing StartPendingInspection Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <PrincipalStartInspection />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
