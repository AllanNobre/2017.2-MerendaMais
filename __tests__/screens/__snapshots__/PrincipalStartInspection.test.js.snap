// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StartPendingInspection Screen renders as expected 1`] = `
<TabViewAnimated
  initialLayout={
    Object {
      "height": 0,
      "width": 0,
    }
  }
  navigationState={
    Object {
      "index": 0,
      "routes": Array [
        Object {
          "key": "first",
          "title": "Pendentes",
        },
        Object {
          "key": "second",
          "title": "Fiscalizados",
        },
        Object {
          "key": "third",
          "title": "Expirados",
        },
      ],
    }
  }
  onIndexChange={[Function]}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderPager={[Function]}
  renderScene={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
